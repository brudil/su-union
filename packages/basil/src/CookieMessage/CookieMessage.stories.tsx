import { storiesOf } from '@storybook/react';
import { CookieMessage } from '../../../website/src/components/CookieMessage/index';

storiesOf('CookieMessage', module).add('default', () => <CookieMessage />);
