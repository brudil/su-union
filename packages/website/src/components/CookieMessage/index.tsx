import React, { useState } from 'react';
import { COLORS, Layers } from '@ussu/common/src/libs/style';
import { type, Typeface, TypeSize } from '@ussu/common/src/libs/style/type';

const CookieMessage: React.FC = () => {
  const [hidden, setHidden] = useState(false);

  if (hidden) {
    return null;
  }

  return (
    <div css={{
      position: 'fixed',
      bottom: '10px',
      zIndex: Layers.GlobalNotice,
      paddingLeft: '1.5rem',
      paddingRight: '1.5rem',
      '& a': {
        color: COLORS.BLACK,
        textDecoration: 'underline',
      },
    }}>
      <div css={{
        ...type(TypeSize.LongPrimer, Typeface.Secondary),
        textAlign: 'center',
        padding: '0.6rem',
        backgroundColor: COLORS.WHITE,
        maxWidth: '300px',
        width: '100%',
        margin: '0 auto',
        boxSizing: 'border-box',
        borderRadius: '2px',
        fontWeight: 600,
      }}>
        {
          'This site uses cookies. By continuing to browse, you agree to our use of cookies on the site. '
        }
        <a href="/cookie-policy">Learn more</a>.
        <div css={{
          paddingTop: '1rem',
        }}>
          <button
            className="Button"
            title="Close"
            onClick={() => setHidden(true)}
            type="button"
            data-testid="close"
          >
            Close
          </button>
        </div>
      </div>
    </div>
  );
};

export default CookieMessage;
