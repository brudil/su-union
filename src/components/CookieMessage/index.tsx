import React from 'react';
import bind from 'bind-decorator';
import { COLORS, Layers } from '~libs/style';
import styled from 'react-emotion';
import { type, Typeface, TypeSize } from '~libs/style/type';
import { cardRaised } from '~libs/style/cards';

interface IProps {}

interface IState {
  hidden: boolean;
}

const MessageContainer = styled.div({
  position: 'fixed',
  bottom: '10px',
  zIndex: Layers.GlobalNotice,
  paddingLeft: '1.5rem',
  paddingRight: '1.5rem',
  '& a': {
    color: COLORS.BLACK,
    textDecoration: 'underline',
  },
});

const MessageBlimp = styled.div(
  {
    ...type(TypeSize.LongPrimer, Typeface.Secondary),
    textAlign: 'center',
    padding: '0.6rem',
    backgroundColor: COLORS.WHITE,
    maxWidth: '300px',
    width: '100%',
    margin: '0 auto',
    boxSizing: 'border-box',
    borderRadius: '2px',
    fontWeight: 600,
  },
  cardRaised,
);

const MessageAction = styled.div({
  paddingTop: '1rem',
});

class CookieMessage extends React.Component<IProps, IState> {
  state = { hidden: false };

  @bind
  handleHide() {
    this.setState({ hidden: true });
  }

  render() {
    if (this.state.hidden) {
      return null;
    }

    return (
      <MessageContainer>
        <MessageBlimp>
          {
            'This site uses cookies. By continuing to browse, you agree to our use of cookies on the site. '
          }
          <a href="/cookie-policy">Learn more</a>.
          <MessageAction>
            <button
              className="Button"
              title="Close"
              onClick={this.handleHide}
              type="button"
            >
              Close
            </button>
          </MessageAction>
        </MessageBlimp>
      </MessageContainer>
    );
  }
}

export default CookieMessage;
