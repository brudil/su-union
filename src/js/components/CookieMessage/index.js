import React from 'react';

class CookieMessage extends React.Component {
  constructor(props) {
    super(props);

    this.state = { hidden: false };

    this.handleHide = this.handleHide.bind(this);
  }

  handleHide() {
    this.setState({ hidden: true });
  }

  render() {
    if (this.state.hidden) {
      return null;
    }

    return (
      <div className="GlobalNotice">
        {
          'This site uses cookies. By continuing to browse, you agree to our use of cookies on the site. '
        }
        <a href="/cookie-policy">Learn more</a>.
        <button
          className="GlobalNotice__close"
          title="Close"
          onClick={this.handleHide}
        >
          <span className="u-h">
            Close
          </span>
        </button>
      </div>
    );
  }
}

export default CookieMessage;
