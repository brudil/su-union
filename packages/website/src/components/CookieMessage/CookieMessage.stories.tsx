import React from 'react';
import { storiesOf } from '@storybook/react';
import CookieMessage from './index';

storiesOf('CookieMessage', module).add('default', () => <CookieMessage />);
