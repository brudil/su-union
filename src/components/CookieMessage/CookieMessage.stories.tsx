import React from 'react';
import { storiesOf } from '@storybook/react';
import CookieMessage from '~components/CookieMessage/index';

storiesOf('CookieMessage', module).add('default', () => <CookieMessage />);
