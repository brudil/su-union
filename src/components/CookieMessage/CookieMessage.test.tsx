import React from 'react';
import { render, fireEvent } from 'react-testing-library';

import CookieMessage from './index';

it('renders a snapshot', () => {
  const { container } = render(<CookieMessage />);
  expect(container).toMatchSnapshot();
});

it('hides on close', () => {
  const { getByTestId, container } = render(<CookieMessage />);
  const closeButton = getByTestId('close');

  fireEvent.click(closeButton);

  expect(container.firstChild).toBeNull();
});
