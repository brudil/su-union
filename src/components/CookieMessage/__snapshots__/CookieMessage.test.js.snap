// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="GlobalNotice"
>
  This site uses cookies. By continuing to browse, you agree to our use of cookies on the site. 
  <a
    href="/cookie-policy"
  >
    Learn more
  </a>
  .
  <button
    className="GlobalNotice__close"
    onClick={[Function]}
    title="Close"
  >
    <span
      className="u-h"
    >
      Close
    </span>
  </button>
</div>
`;
